common/types.d.ts

```diff
@@ -115,3 +115,20 @@ export type GenerateCodeOptions = {
   actions: Steps;
   isProject: boolean;
 };
+export interface IElectronAPI {
+  exportScript: (string) => Promise<boolean>;
+  recordJourney: (url: string) => Promise<void>;
+  stopRecording: () => void;
+  pauseRecording: () => Promise<void>;
+  resumeRecording: () => Promise<void>;
+  onActionGenerated: (
+    callback: (event: IpcRendererEvent, actions: ActionContext[]) => void
```

Comment:
```suggestion
    callback: ActionGeneratedCallback
```

I mention this further down in the review, but I think it might be nice to do something like this:

```typescript
type ActionGeneratedCallback = (event: IpcRendererEvent, actions: ActionContext[]) => void;
```

Then we can reference this type in the implementation of `onActionGenerated`, and where we set it up in `preload.ts`. It gives a little shorthand for defining the argument type.

Reply:
Thanks for this comment! While applying your suggestion, I've observed an interesting DX case I'd like your opinion on. I usually prefer the way suggested, but this time it seems like it is quite useful to have ad-hoc type definition when implementing this function elsewhere;  If I define the type for the callback and hover when using the function, it will make me go over to check the details of `ActionGeneratedCallback`type definition:

<img width="800" alt="image" src="https://user-images.githubusercontent.com/16660866/217465582-76fe718e-6d26-477b-a457-cd99696baff7.png">

But when it's defined inline, I can check the type definition of the callback straight away without hopping the definition chains:

<img width="826" alt="image" src="https://user-images.githubusercontent.com/16660866/217465700-45b6144f-4d28-4b4e-90fc-9a5530da32a3.png">

Also, I noticed that I didn't need to define the argument type again in `preload.ts` so we don't need to reference this type again in there.

WDYT @justinkambic ?!

Reply:
Sure, that is fine with me as well. Given that this signature is only expected a handful of times in this case I think it's an ok tradeoff.

If, for example, we were making a library where we were expecting to reference this signature an arbitrary amount of times, I'd say we should probably define a custom type. It can have added DX burden if you need to drill into the types, but I think it's generally cleaner when reading the code. I'm curious what happens if you hover on `callback` in this case, does it only show `ActionGeneratedCallback` as well?

Reply:
When you hover on `callback` (I renamed it to `listener` here), it shows type definition that I specified for the callback, but as soon as I put the wrong types and try to use it in `electronAPI.addActionGeneratedListner`, it complains:
https://gyazo.com/5759b1b9c960bcde61f4c39ab8e99669